modules/assim.sequential/R/sda_plotting.R

```diff
@@ -142,16 +141,16 @@ postana.timeser.plotting.sda<-function(settings, t, obs.times, obs.mean, obs.cov
   Y.order <- sapply(colnames(FORECAST[[t]]),agrep,x=colnames(Ybar),max=2,USE.NAMES = F)%>%unlist
   Ybar <- Ybar[,Y.order]
   YCI <- t(as.matrix(sapply(obs.cov[t1:t], function(x) {
-    if (is.null(x)) {
+    if (is.na(x)) {
       rep(NA, length(names.y))
-    }
-    sqrt(diag(x))
+    }else{
+    sqrt(diag(x))}
   })))
   
   Ybar[is.na(Ybar)]<-0
   YCI[is.na(YCI)]<-0
   
-  YCI <- YCI[,Y.order]
+  YCI <- YCI[,c(Y.order)]
```

Comment:
Why the `c` here? Is it to implicitly convert `Y.order` from a matrix to a vector, or something else? If so, can you add a brief comment explaining that here?